### ISSUE TITLE
[Redshift] Parsing error when defining multiple columns as foreign key constraints

### ISSUE THREAD

**Issue by maoxingda** [sqlfluff/sqlfluff#5592]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The parser was unable to parse multiple columns foreign key constraints.

### Expected Behaviour

SQLFluff parse the file correctly.

### Observed Behaviour

I created a file that contains the following:

```sql
create table public.t1 (c1 int, c2 int, unique (c1, c2));

create table public.t2
(
    c1 int,
    c2 int,
    foreign key (c1, c2) references public.t1 (c1, c2)
);
```

After that, I ran this command:

```shell
sqlfluff parse -d redshift test.sql
```

Output:

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_table_statement:
[L:  1, P:  1]      |            keyword:                                          'create'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'table'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            table_reference:
[L:  1, P: 14]      |                naked_identifier:                             'public'
[L:  1, P: 20]      |                dot:                                          '.'
[L:  1, P: 21]      |                naked_identifier:                             't1'
[L:  1, P: 23]      |            whitespace:                                       ' '
[L:  1, P: 24]      |            bracketed:
[L:  1, P: 24]      |                start_bracket:                                '('
[L:  1, P: 25]      |                [META] indent:
[L:  1, P: 25]      |                column_reference:
[L:  1, P: 25]      |                    naked_identifier:                         'c1'
[L:  1, P: 27]      |                whitespace:                                   ' '
[L:  1, P: 28]      |                data_type:
[L:  1, P: 28]      |                    keyword:                                  'int'
[L:  1, P: 31]      |                comma:                                        ','
[L:  1, P: 32]      |                whitespace:                                   ' '
[L:  1, P: 33]      |                column_reference:
[L:  1, P: 33]      |                    naked_identifier:                         'c2'
[L:  1, P: 35]      |                whitespace:                                   ' '
[L:  1, P: 36]      |                data_type:
[L:  1, P: 36]      |                    keyword:                                  'int'
[L:  1, P: 39]      |                comma:                                        ','
[L:  1, P: 40]      |                whitespace:                                   ' '
[L:  1, P: 41]      |                table_constraint:
[L:  1, P: 41]      |                    keyword:                                  'unique'
[L:  1, P: 47]      |                    whitespace:                               ' '
[L:  1, P: 48]      |                    bracketed:
[L:  1, P: 48]      |                        start_bracket:                        '('
[L:  1, P: 49]      |                        [META] indent:
[L:  1, P: 49]      |                        column_reference:
[L:  1, P: 49]      |                            naked_identifier:                 'c1'
[L:  1, P: 51]      |                        comma:                                ','
[L:  1, P: 52]      |                        whitespace:                           ' '
[L:  1, P: 53]      |                        column_reference:
[L:  1, P: 53]      |                            naked_identifier:                 'c2'
[L:  1, P: 55]      |                        [META] dedent:
[L:  1, P: 55]      |                        end_bracket:                          ')'
[L:  1, P: 56]      |                [META] dedent:
[L:  1, P: 56]      |                end_bracket:                                  ')'
[L:  1, P: 57]      |    statement_terminator:                                     ';'
[L:  1, P: 58]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  2, P:  1]      |        word:                                                 'create'
[L:  2, P:  7]      |        whitespace:                                           ' '
[L:  2, P:  8]      |        word:                                                 'table'
[L:  2, P: 13]      |        whitespace:                                           ' '
[L:  2, P: 14]      |        word:                                                 'public'
[L:  2, P: 20]      |        dot:                                                  '.'
[L:  2, P: 21]      |        word:                                                 't2'
[L:  2, P: 23]      |        newline:                                              '\n'
[L:  3, P:  1]      |        start_bracket:                                        '('
[L:  3, P:  2]      |        newline:                                              '\n'
[L:  4, P:  1]      |        whitespace:                                           '    '
[L:  4, P:  5]      |        word:                                                 'c1'
[L:  4, P:  7]      |        whitespace:                                           ' '
[L:  4, P:  8]      |        word:                                                 'int'
[L:  4, P: 11]      |        comma:                                                ','
[L:  4, P: 12]      |        newline:                                              '\n'
[L:  5, P:  1]      |        whitespace:                                           '    '
[L:  5, P:  5]      |        word:                                                 'c2'
[L:  5, P:  7]      |        whitespace:                                           ' '
[L:  5, P:  8]      |        word:                                                 'int'
[L:  5, P: 11]      |        comma:                                                ','
[L:  5, P: 12]      |        newline:                                              '\n'
[L:  6, P:  1]      |        whitespace:                                           '    '
[L:  6, P:  5]      |        word:                                                 'foreign'
[L:  6, P: 12]      |        whitespace:                                           ' '
[L:  6, P: 13]      |        word:                                                 'key'
[L:  6, P: 16]      |        whitespace:                                           ' '
[L:  6, P: 17]      |        start_bracket:                                        '('
[L:  6, P: 18]      |        word:                                                 'c1'
[L:  6, P: 20]      |        comma:                                                ','
[L:  6, P: 21]      |        whitespace:                                           ' '
[L:  6, P: 22]      |        word:                                                 'c2'
[L:  6, P: 24]      |        end_bracket:                                          ')'
[L:  6, P: 25]      |        whitespace:                                           ' '
[L:  6, P: 26]      |        word:                                                 'references'
[L:  6, P: 36]      |        whitespace:                                           ' '
[L:  6, P: 37]      |        word:                                                 'public'
[L:  6, P: 43]      |        dot:                                                  '.'
[L:  6, P: 44]      |        word:                                                 't1'
[L:  6, P: 46]      |        whitespace:                                           ' '
[L:  6, P: 47]      |        start_bracket:                                        '('
[L:  6, P: 48]      |        word:                                                 'c1'
[L:  6, P: 50]      |        comma:                                                ','
[L:  6, P: 51]      |        whitespace:                                           ' '
[L:  6, P: 52]      |        word:                                                 'c2'
[L:  6, P: 54]      |        end_bracket:                                          ')'
[L:  6, P: 55]      |        newline:                                              '\n'
[L:  7, P:  1]      |        end_bracket:                                          ')'
[L:  7, P:  2]      |        semicolon:                                            ';'
[L:  7, P:  3]      |    newline:                                                  '\n'
[L:  8, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'create table
                       | public.t2\n(\n    c1 int,\n   ...'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
```

### How to reproduce

```shel
sqlfluff parse -d redshift test.sql
```

### Dialect

redshift

### Version

sqlfluff, version 3.0.0a4

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
